src/app.jsx

```diff
@@ -148,15 +157,40 @@ class Application extends React.Component {
                         for (const container of reply || []) {
                             container.isSystem = system;
                             copyContainers[container.Id + system.toString()] = container;
-                            if (container.State === "running")
-                                this.updateContainerStats(container.Id, system);
                         }
+                        // containersDetails, containersStats are updated/deleted on other events
+                        return {
+                            containers: copyContainers,
+                            [system ? "systemContainersLoaded" : "userContainersLoaded"]: true,
+                        };
+                    });
+                })
+                .catch(e => console.log(e));
+    }
 
+    initContainers(system) {
+        client.getContainers(system)
+                .then(reply => {
+                    this.setState(prevState => {
+                        const copyContainers = {};
+                        Object.entries(prevState.containers || {}).forEach(([id, container]) => {
+                            if (container.isSystem !== system)
+                                copyContainers[id] = container;
+                        });
+                        for (const container of reply || []) {
+                            container.isSystem = system;
+                            copyContainers[container.Id + system.toString()] = container;
+                        }
                         return {
                             containers: copyContainers,
                             [system ? "systemContainersLoaded" : "userContainersLoaded"]: true,
                         };
                     });
+                    for (const container of reply || []) {
```

Comment:
So the new `initContainers` differes from `updateContainersAfterEvent` only in the last 5 lines, right?
In that case I would just give `updateContainersAfterEvent` additional argument and made these line conditional on this argument.

Reply:
Yeah you're right. I just thought they got different meanings, so choose to leave it this way.